util/files/tar.go

```diff
@@ -0,0 +1,169 @@
+package files
+
+import (
+	"archive/tar"
+	"bufio"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type tgz struct {
+	srcPath    string
+	exclusions []string
+	tarWriter  *tar.Writer
+}
+
+// Tgz will iterate over all files found in srcPath compressing them with gzip
+// and archiving with Tar. Will invoke every given writer while generating the tgz.
+// This is useful to generate checksums. Will exclude files matching the exclusions
+// list blob.
+func Tgz(srcPath string, exclusions []string, writers ...io.Writer) error {
+	if _, err := os.Stat(srcPath); err != nil {
+		return fmt.Errorf("error inspecting srcPath %q: %w", srcPath, err)
+	}
+
+	mw := io.MultiWriter(writers...)
+
+	gzw := gzip.NewWriter(mw)
+	defer gzw.Close()
+
+	tw := tar.NewWriter(gzw)
+	defer tw.Close()
+
+	t := &tgz{
+		srcPath:    srcPath,
+		exclusions: exclusions,
+		tarWriter:  tw,
+	}
+	return filepath.Walk(srcPath, t.tgzFile)
+}
+
+// Untgz will loop over the tar reader creating the file structure at dstPath
+func Untgz(dstPath string, r io.Reader) error {
+
+	gzr, err := gzip.NewReader(r)
+	if err != nil {
+		return fmt.Errorf("error reading file: %w", err)
+	}
+	defer gzr.Close()
+
+	tr := tar.NewReader(gzr)
+
+	for {
+		header, err := tr.Next()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return fmt.Errorf("error while iterating on tar reader: %w", err)
```

Comment:
I'm not sure what kind of information tar might put in the error message. Maybe we should return a simple message and just log the details? I doubt that an error occurring here is likely to be fixable by the end user.

Reply:
`Untgz` is a general purpose function. It shouldn't care if it is dealing with sensitive data or not. It is the caller's responsibility to decide what to do with the error message.